aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -267,16 +267,16 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
     try {
       TableOperations ops = newTableOps(identifier);
       TableMetadata lastMetadata = ops.current();
+      if (purge && lastMetadata != null) {
+        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        LOG.info("Glue table {} data purged", identifier);
+      }
       glue.deleteTable(DeleteTableRequest.builder()
           .catalogId(awsProperties.glueCatalogId())
           .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
           .name(identifier.name())
           .build());
       LOG.info("Successfully dropped table {} from Glue", identifier);
-      if (purge && lastMetadata != null) {
-        CatalogUtil.dropTableData(ops.io(), lastMetadata);
-        LOG.info("Glue table {} data purged", identifier);
-      }
```

Comment:
ah interesting, I overlooked this aspect. Does that mean when user has LF enabled, we must purge the table before dropping the table in Glue, otherwise it cannot get temporary S3 credential?

If that's the case, we should make delete table in the finally block, so that purge failure does not impact actual table drop.